src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -13678,6 +13774,14 @@ int Assembler::simd_prefix_and_encode(XMMRegister dst, XMMRegister nds, XMMRegis
   }
 }
 
+bool Assembler::is_demotable(bool no_flags, int dst_enc, int nds_enc, int src_enc) {
+  return (!no_flags && dst_enc == nds_enc);
+}
+
+bool Assembler::is_demotable(bool no_flags, int dst_enc, int nds_enc) {
+  return (!no_flags && dst_enc == nds_enc);
+}
+
```

Comment:
Just a heads up, we are missing a case where dst_enc can be equal to src_enc; in that case we can still demote EVEX to REX/REX2 encoding along with a change in primary opcode if needed.